feat(specs): Add EIP-7981 implementation

nerolation opened the pull request:
First draft implementation of EIP-7981, which is going to be proposed for Glamsterdam.

gurukamath commented on src/ethereum/osaka/transactions.py:
```diff
@@ -568,6 +584,40 @@ def validate_transaction(tx: Transaction) -> Tuple[Uint, Uint]:
+    
+    for access in access_list:
+        # Count bytes in address (20 bytes)
+        for byte in access.account:
```
Counting zero and non-zero bytes is being used in a couple of other places as well, including for call data in `calculate_intrinsic_cost`. Perhaps it would be a good idea to refactor it into a function in the `ethereum.utils.byte` module.

nerolation replied:
yeah agree, makes sense

fselmo committed:
feat(specs): create Amsterdam fork from Osaka

nerolation committed:
Implement EIP-7981: Repricing Access Lists

Add additional gas costs for access list data bytes as specified in EIP-7981.

- Add count_bytes utility function for zero/non-zero byte counting
- Add EIP-7981 gas cost constants for access list bytes
- Add count_access_list_bytes function to count bytes in access lists
- Modify calculate_intrinsic_cost to include EIP-7981 access list data costs
- Update function documentation to reference EIP-7981

Maintains compatibility with existing EIP-2930 access list functionality.

gurukamath commented:
@nerolation Does #2144 supersede this? If yes, I think this can be closed. wdyt?
